core/src/main/scala/persistentscheduler/PersistentScheduler.scala

```diff
@@ -1,52 +1,57 @@
 
 import scala.collection.JavaConverters._
 import scala.compat.java8.OptionConverters._
+import scala.concurrent.Future
 import scala.concurrent.duration._
 
 object PersistentScheduler {
 
   sealed trait Result
-  case class Info(self: ActorRef) extends Result
+  case class Info(self: ActorRef)              extends Result
   case class SubscribedActorRef(ref: ActorRef) extends Result
-  case class Scheduled(event: TimedEvent) extends Result
+  case class Scheduled(event: TimedEvent)      extends Result
   case class RemovedEventsByReference(eventType: String, reference: String)
   case class FoundEventsByReference(eventType: String, reference: String, events: List[TimedEvent])
 
   sealed trait Request
-  case class IsAlive() extends Request
-  case class Schedule(event: TimedEvent) extends Request
-  case class SubscribeActorRef(subscriber: ActorRef, eventType: String) extends Request
+  case class IsAlive()                                                     extends Request
+  case class Schedule(event: TimedEvent)                                   extends Request
+  case class SubscribeActorRef(subscriber: ActorRef, eventType: String)    extends Request
   case class RemoveEventsByReference(eventType: String, reference: String) extends Request
-  case class PublishEvent(event: TimedEvent) extends Request
-  case class CheckPersistenceForEvents() extends Request
-  case class FindEventsByReference(eventType: String, reference: String) extends Request
+  case class PublishEvent(event: TimedEvent)                               extends Request
+  case class CheckPersistenceForEvents()                                   extends Request
+  case class FindEventsByReference(eventType: String, reference: String)   extends Request
 
   case class State(subscriptions: Set[Subscription])
   case class Subscription(eventType: String, subscriber: ActorRef)
 
-  def props(persistence: SchedulerPersistence) = Props(new PersistentScheduler(persistence))
+  def props(persistence: SchedulerPersistence)(implicit timeout: Timeout) = Props(new PersistentScheduler(persistence))
 
-  def apply(persistence: SchedulerPersistence): PersistentScheduler = new PersistentScheduler(persistence)
+  def apply(persistence: SchedulerPersistence)(implicit timeout: Timeout): PersistentScheduler =
+    new PersistentScheduler(persistence)
 }
 
-class PersistentScheduler(persistence: SchedulerPersistence) extends Actor
-  with AkkaScheduler {
+class PersistentScheduler(persistence: SchedulerPersistence)(implicit timeout: Timeout)
+    extends Actor
+    with AkkaScheduler {
 
   implicit val ec = context.dispatcher
 
-  var subscriptions: Set[Subscription] = Set()
-  var nextEvent: Option[TimedEvent] = None
+  var subscriptions: Set[Subscription]     = Set()
+  var nextEvent: Option[TimedEvent]        = None
   var nextCancellable: Option[Cancellable] = None
 
   override def receive: Receive = {
     case IsAlive() =>
-      sender() ! Info(self)
+      sender() ? Info(self)
```

Comment:
I think you are mixing something up here. You don't need to ask the sender for something, because the sender (PersistentSchedulerExtension.scala) actually wants something from you and you have to reply to him, so the current implementation with _!_ is correct.

https://doc.akka.io/docs/akka/current/actors.html#send-messages